controller/node_controller.go

```diff
@@ -1027,60 +1031,68 @@ func (nc *NodeController) syncMultipathd(node *longhorn.Node, namespaces []lhtyp
 	node.Status.Conditions = types.SetCondition(node.Status.Conditions, longhorn.NodeConditionTypeMultipathd, longhorn.ConditionStatusTrue, "", "")
 }
 
-func (nc *NodeController) syncNFSClientVersion(kubeNode *corev1.Node, node *longhorn.Node, namespaces []lhtypes.Namespace) {
-	kernelVersion := kubeNode.Status.NodeInfo.KernelVersion
-	nfsClientVersions := []string{"CONFIG_NFS_V4_2", "CONFIG_NFS_V4_1", "CONFIG_NFS_V4"}
+func (nc *NodeController) checkModulesLoaded(kubeNode *corev1.Node, node *longhorn.Node, namespaces []lhtypes.Namespace) {
+	checkingModules := map[string]string{
+		"CONFIG_DM_CRYPT": "dm_crypt",
+	}
 
-	nsexec, err := lhns.NewNamespaceExecutor(lhtypes.ProcessNone, lhtypes.HostProcDirectory, namespaces)
+	kernelConfigPath := kernelConfigFilePathPrefix + kubeNode.Status.NodeInfo.KernelVersion
+	configContent, err := lhio.ReadFileContent(kernelConfigPath)
 	if err != nil {
-		node.Status.Conditions = types.SetCondition(node.Status.Conditions, longhorn.NodeConditionTypeNFSClientInstalled, longhorn.ConditionStatusFalse,
-			string(longhorn.NodeConditionReasonNamespaceExecutorErr),
-			fmt.Sprintf("Failed to get namespace executor: %v", err.Error()))
+		node.Status.Conditions = types.SetCondition(node.Status.Conditions, longhorn.NodeConditionTypeModulesLoaded, longhorn.ConditionStatusFalse,
+			string(longhorn.NodeConditionReasonKernelConfigIsNotFound),
+			fmt.Sprintf("Unable to get %v content to check if %v are enabled: %v", kernelConfigPath, checkingModules, err.Error()))
+		return
+	}
```

Comment:
If the config file does not exist, we cannot continue to check the kernel modules using `lsmod` because of the error out.

Reply:
So we will want to use `lsmod` or `kmod` commands to check if the modules are loaded even though the kernel config file does not exist, right?

Reply:
Modified. Thanks